flutter/android/app/build.gradle

```diff
@@ -44,7 +44,7 @@ android {
     defaultConfig {
         applicationId "org.mlcommons.android.mlperfbench"
         minSdkVersion 16
-        targetSdkVersion 31
+        targetSdkVersion 30
```

Comment:
With target SDK v30 we will not be able to publish the app. See
https://developer.android.com/google/play/requirements/target-sdk

Reply:
but seems like targetSdkVersion 31 is somehow creating issues with flutter. The app is not able to read the files from /system/lib64 folder. So, the system dependent libraries like libcdsprpc.so are not found when loading libqtibackend.so.

My suggestion is to have this changed to 30 and find a solution for the library load issue separately.

Reply:
I found the following info in Android documentation:
>Applications that target SDK version 31 (Android 12) or higher must explicitly specify their native shared library dependencies by using the <uses-native-library> tag in the app manifest. If any part of the requested library doesn't exist on the device, the app isn't installed. When the apps are installed, they're provided with only the native shared libraries that they've requested. This means that apps can't access native shared libraries that don't appear in the app manifest.

https://source.android.com/devices/tech/config/namespaces_libraries

It seems to me that if we want to create a unified app for all backends we will have to include all vendor-specific libraries into the app, or else the app won't even install anywhere.

Reply:
Downgrade to v30 is not really a choice if we want to publish the app. We can try what @d-uzlov suggest.

Reply:
looking into it if doing that works.

Reply:
I think we can merge this for now and update the SDK target version later, after we find the solution.

Reply:
Please discuss it in the mobile engineering meeting how we want to handle it.

Reply:
It seems there are no issues with specifying list of vendor-specific native libraries after all.
There is a flag to mark dependency as optional, and with the last commit from @AasheeshT I'm able to install the app on my phone which doesn't have specified Qualcomm native lib.

Reply:
Also worked with my test device.